packages/swap/src/utils/ServiceError.ts

```diff
@@ -5,30 +5,43 @@ type IgnoredField =
   | 'prepareStackTrace'
   | 'stackTraceLimit';
 
+type ErrorCode = 'not-found' | 'bad-request' | 'internal-error' | 'rpc-error' | 'invalid-amount';
```

Comment:
another possibility would be having no general code (like `bad-request`), and only set the code for specific errors. but i dont think i like that

Reply:
the problem with the current approach is that we cannot really change from a general code (`bad-request`) to a more specific one (`invalid-address`) because this will break consumers. 

we could add a third property (`detail`) which will be set for specific errors (eg `invalid-amount`, `invalid-address`) only? but this feels overly complicated

Reply:
TIL that there is a IETF RFC for exposing errors: https://datatracker.ietf.org/doc/html/rfc7807
will check if this makes sense in our case

Reply:
there even is a new RPC from 2023 that obsoletes the old one: https://datatracker.ietf.org/doc/html/rfc9457
but it seems like this standard is not widely used and the format feels specific to errors returned from an api (which makes sense for the rpc)

Reply:
i am leaning to use the following format:
```
{
  type: 'NOT_FOUND' | 'BAD_REQUEST';
  code: 'INVALID_AMOUNT' | undefined,
  message: 'human readable explanation'
}
```

this would allow use to introduce specific error codes later on without breaking any consumer. (eg. adding a `INVALID_ADDRESS` code)

Reply:
this would be kinda similar to what stripe/twilio does:
- https://docs.stripe.com/api/errors
- https://www.twilio.com/docs/usage/troubleshooting/debugging-your-application#debugging-calls-to-the-rest-api

Reply:
Should we maybe just use numeric codes instead of keywords? Using keywords is an approach prone to collisions and the chances that we might need to change an old code (keyword) to be more/less explicit to accomodate a new error code is higher.
If we use numeric codes we can simply guarantee they wont change while we can still change the keywords/description to provide more context.

Reply:
Imagine using `INVALID_AMOUNT` at this point for indicating that there was an issue with the swap amount. What if tomorrow we add another endpoint that requires amounts to be provided. At that point, the `INVALID_AMOUNT` might apply to both cases, and it will not be possible to change this old code (keyword).

Reply:
i agree that this could lead to keywords being less explicit than they could be bc of backwards compatibility, a number would allow you to clarify the "meaning" later on. but this also means that the number itself does not give you an idea about the error (without consulting documentation). to solve that you could provide constants with meaningful names, but then you kinda end up with the same problem of not being able to change them easily 🙈